src/target/arc_ocd.c

```diff
@@ -61,11 +61,26 @@ int arc_ocd_poll(struct target *target)
 			target->state = TARGET_RUNNING;
 		}
 	} else {
-		if ((target->state == TARGET_RUNNING) ||
-			(target->state == TARGET_RESET)) {
+		/* In some cases JTAG status register indicates that
+		*  processor is in halt mode, but processor is still running.
+		*  We check halt bit of AUX STATUS32 register for setting correct state
+		*/
+		if (target->state == TARGET_RUNNING) {
+			CHECK_RETVAL(arc_jtag_read_aux_reg_one(&arc32->jtag_info, AUX_STATUS32_REG, &value));
+			if ((value & SET_CORE_HALT_BIT)) {
+				LOG_DEBUG("ARC core in halt state, ");
+
+				target->state = TARGET_HALTED;
+				CHECK_RETVAL(arc_dbg_debug_entry(target));
+				target_call_event_callbacks(target, TARGET_EVENT_DEBUG_HALTED);
+			}
+			else {
+				 LOG_DEBUG("discrepancy of STATUS32(0) HALT bit and ARC_JTAG_STAT_RU, target is still running");
```

Comment:
Start message with a capital letter. Also, I would recommend writing `STATUS32[0]` - with square brackets, for consistency with PRM (and I presume rest of the ARC code, because *if* there are similar references in code, I would have done them with square branckes).